indico/modules/categories/templates/display/sidebar.html

```diff
@@ -1,89 +1,177 @@
+    {% set folder_name = attachment.title %}
+    <li>
+        <details open>
+            <summary aria-label="folder {{ attachment.title }}">
+                {{ attachment.title }}
+            </summary>
+            <ul>
+                {% for child in attachment.attachments %}
+                    {{ _render_attachment_file(child) }}
+                {% endfor %}
+            </ul>
+        </details>
+    </li>
+{% endmacro %}
 
 {% macro render_attachments(category) %}
     {%- set can_manage = category.can_manage(session.user) -%}
     {%- if category.attachment_count or can_manage %}
-        <div class="title">
-            <h2 class="icon-attachment">{% trans %}Materials{% endtrans %}</h2>
-            {% if can_manage %}
-                <div class="actions">
-                    <button class="i-button text-color subtle icon-edit js-manage-attachments"
-                            title="{% trans %}Manage materials{% endtrans %}"
-                            data-title="{% trans title=category.title %}Manage category materials{% endtrans %}"
-                            data-locator="{{ category.locator|tojson|forceescape }}"
-                            data-attachment-editor
-                            data-reload-on-change></button>
+        <section id="material-list" aria-labelledby="materials">
+            <header>
+                <h2 id="materials">{% trans %}Materials{% endtrans %}</h2>
+                {% if can_manage %}
+                    <div class="actions">
+                        <button class="i-button text-color subtle icon-edit js-manage-attachments"
+                                title="{% trans %}Manage materials{% endtrans %}"
+                                data-title="{% trans title=category.title %}Manage category materials{% endtrans %}"
+                                data-locator="{{ category.locator|tojson|forceescape }}"
+                                data-attachment-editor
+                                data-reload-on-change></button>
+                    </div>
+                {% endif %}
+            </header>
+            <ul>
+                {% set attachments = category.attached_items %}
+                {% if attachments %}
+                    {% for attachment in attachments.files %} {{ _render_attachment_file(attachment) }}{% endfor %}
+                    {% for attachment in attachments.folders %}{{ _render_attachment_folder(attachment) }}{% endfor %}
+                {% else %}
+                    <p class="empty-list-notice">{% trans %}There are no materials yet.{% endtrans %}</p>
+                {% endif %}
+            </ul>
+        </section>
+
+        <ind-attachment-preview-modal>
+            <dialog>
+                <div>
+                    <div class="titlebar">
+                        <h2>{% trans %}Attachment preview{% endtrans %}</h2>
+                        <button value="close"><span>{% trans %}Close{% endtrans %}</span></button>
```

Comment:
Still not a huge fan of "abusing" a plain span to indicate visually-hidden content here - without knowing the CSS anyone would think this this span is unnecessary...

Reply:
I see what you mean. Sadly, we don't really have a good solution in this case because the icon is inserted via CSS and becomes part of the text content of the element. We therefore have no good way of distinguishing between the text and the icon other than wrapping it in an element that has no semantics.

Now, you could technically somehow mark that element, but then you're marking it for **presentational** reasons, rather than semantics, which is not HTML's job. In fact, I would say presentational markup takes us straight back to the dark ages of HTML 3.2 and is a proper abuse, without quotes.